packages/governance/src/typeGuards.js

```diff
@@ -0,0 +1,103 @@
+// @ts-check
+
+import { M } from '@agoric/store';
+import { TimestampShape } from '../../SwingSet/src/vats/timer/typeGuards';
+
+export const ChoiceMethodShape = M.or('unranked', 'order');
+export const QuorumRuleShape = M.or('majority', 'no_quorum', 'all');
+export const ElectionTypeShape = M.or(
+  'param_change',
+  'election',
+  'survey',
+  'api_invocation',
+  'offer_filter',
+);
+
+export const TimerShape = M.handle('timer');
+export const InstanceShape = M.handle('instance');
+
+export const ClosingRuleShape = harden({
+  timer: M.eref(TimerShape),
+  deadline: TimestampShape,
+});
+
+// all the strings that will be in the filter after passing
+export const YesOfferFilterPositionShape = M.arrayOf(M.string());
+export const NoOfferFilterPositionShape = harden({
+  dontUpdate: M.arrayOf(M.string()),
+});
+export const OfferFilterPositionsShape = [
+  YesOfferFilterPositionShape,
+  NoOfferFilterPositionShape,
+];
+export const OfferFilterIssueShape = harden({
+  strings: M.arrayOf(M.string()),
+});
+export const OfferFilterQuestionSpecShape = harden({
+  method: ChoiceMethodShape,
+  issue: OfferFilterIssueShape,
+  positions: OfferFilterPositionsShape,
+  electionType: ElectionTypeShape,
+  maxChoices: M.eq(1n),
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoOfferFilterPositionShape,
+});
+
+// keys are parameter names, values are proposed values
+export const ParamChangesSpecShape = M.recordOf(M.string(), M.any());
+export const YesParamChangesPositionShape = ParamChangesSpecShape;
+export const NoParamChangesPositionShape = harden({
+  noChange: M.arrayOf(M.string()),
+});
+export const ParamChangesPositionsShape = [
```

Comment:
```suggestion
export const ParamChangesPositionsShape = harden([
```
and several other places

Reply:
done